packages/evm/src/provider/rpc.ts

```diff
@@ -149,11 +218,9 @@ export class JsonRPCWallet extends BaseProviderWallet {
 
   public async changeNetwork(chainOrRpc: RpcType | RpcId = RpcId.ETH): Promise<void> {
     if (typeof chainOrRpc === 'number') {
-      if (chainOrRpc === RpcId.ETH) {
-        await this.provider.send('wallet_switchEthereumChain', [{ chainId: '0x1' }])
-      } else await this.provider.send('wallet_addEthereumChain', [ChainData[chainOrRpc]])
+      await this.provider.send('wallet_switchEthereumChain', [{ chainId: ChainData[chainOrRpc].chainId }])
     } else {
-      await this.provider.send('wallet_addEthereumChain', [chainOrRpc])
+      await this.provider.send('wallet_switchEthereumChain', [{ chainId: chainOrRpc }])
     }
   }
```

Comment:
Simplified this to always use `wallet_switchEthereumChain` as `wallet_addEthereumChain` was giving problems. 

cc. @philogicae